File: backend/Raporlar/KonaklamaKisiSayisi/Dto/KonaklamaKisiSayisiRaporDto.cs
namespace STYS.Raporlar.KonaklamaKisiSayisi.Dto;

public class KonaklamaKisiSayisiRaporDto
{
    public int TesisId { get; set; }

    public string? TesisAdi { get; set; }

    public int Ay { get; set; }

    public string AyAdi { get; set; } = "";

    public int BaslangicYil { get; set; }

    public int BitisYil { get; set; }

    public string Baslik { get; set; } = "";

    public List<KonaklamaKisiSayisiOdaDto> Odalar { get; set; } = [];

    public List<KonaklamaKisiSayisiYilSatiriDto> Yillar { get; set; } = [];
}

File: frontend/src/app/pages/raporlar/konaklama-kisi-sayisi/konaklama-kisi-sayisi-rapor.dto.ts
export interface KonaklamaKisiSayisiOdaDto {
    odaId: number;
    odaNo: string;
    odaTipiAdi: string | null;
    kapasite: number;
}

export interface KonaklamaKisiSayisiHucreDto {
    odaId: number;
    odaNo: string;
    kisiSayisi: number;
}

export interface KonaklamaKisiSayisiYilSatiriDto {
    yil: number;
    hucreler: KonaklamaKisiSayisiHucreDto[];
    toplamKisiSayisi: number;
}

export interface KonaklamaKisiSayisiRaporDto {
    tesisId: number;
    tesisAdi: string | null;
    ay: number;
    ayAdi: string;
    baslangicYil: number;
    bitisYil: number;
    baslik: string;
    odalar: KonaklamaKisiSayisiOdaDto[];
    yillar: KonaklamaKisiSayisiYilSatiriDto[];
}
